springcloud-streams/src/main/java/org/axonframework/extensions/springcloud/eventhandling/SpringStreamMessageProcessor.java

```diff
@@ -0,0 +1,207 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.extensions.springcloud.eventhandling;
+
+import org.axonframework.common.AxonConfigurationException;
+import org.axonframework.common.Registration;
+import org.axonframework.eventhandling.EventBus;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.extensions.springcloud.eventhandling.converter.SpringMessageConverter;
+import org.axonframework.messaging.SubscribableMessageSource;
+import org.springframework.integration.handler.AbstractMessageProducingHandler;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.support.GenericMessage;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+import static java.util.Collections.singletonList;
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+
+/**
+ * Initialize the adapter to publish all incoming events to the subscribed processors. Note that this instance should
+ * be registered as a consumer of a Spring Message Channel.
+ *
+ * @author Mehdi chitforoosh
+ * @since 4.5
+ */
+public class SpringStreamMessageProcessor extends AbstractMessageProducingHandler implements SubscribableMessageSource<EventMessage<?>> {
+
+    private final SubscribableMessageSource<EventMessage<?>> messageSource;
+    private final CopyOnWriteArrayList<Consumer<List<? extends EventMessage<?>>>> messageProcessors = new CopyOnWriteArrayList<>();
+    private Predicate<? super EventMessage<?>> filter;
+    private final SpringMessageConverter converter;
+
+    /**
+     * Instantiate a {@link SpringStreamMessageProcessor} based on the fields contained in the {@link SpringStreamMessageProcessor.Builder}.
+     * The {@link SubscribableMessageSource} is a <b>hard requirement</b> and thus should be provided.
+     * The {@link SpringMessageConverter} is a <b>hard requirement</b> and thus should be provided.
+     * <p>
+     * Will validate that the {@link SubscribableMessageSource} and {@link SpringMessageConverter} and {@link EventBus} are not {@code null},
+     * and will throw an {@link AxonConfigurationException} if for either of them this holds.
+     *
+     * @param builder the {@link SpringStreamMessageProcessor.Builder} used to instantiate a {@link SpringStreamMessageProcessor} instance
+     */
+    protected SpringStreamMessageProcessor(SpringStreamMessageProcessor.Builder builder) {
+        builder.validate();
+        this.filter = builder.filter;
+        this.messageSource = builder.messageSource;
+        this.converter = builder.converter;
+        if (builder.eventBus != null) {
+            this.messageProcessors.addAll(singletonList(builder.eventBus::publish));
```

Comment:
I wonder if we should add EventBus by default here? @smcvb what do you think? In my opinion, we could leave this decision to the users of the API, but I am afraid that in that case, we are missing something like `MultiSubscribableMessageSource.java` allowing them to subscribe to multiple sources: eventBus/eventStore, streamSource, ...